models/points.go

```diff
@@ -232,11 +232,32 @@ func parsePoint(buf []byte, defaultTime time.Time, precision string) (Point, err
 		if err != nil {
 			return nil, err
 		}
-		pt.time = time.Unix(0, ts*pt.GetPrecisionMultiplier(precision)).UTC()
+		pt.time, err = SafeCalcTime(ts, precision)
```

Comment:
For the case that the precision is in nanoseconds the only overhead is the function call which the compiler is hopefully in-lining anyway. The below result show that there isn't a measurable difference between master and this branch.

Master benchmarks for models:

```
BenchmarkMarshal-8                        300000              5026 ns/op
BenchmarkParsePointNoTags-8              2000000               612 ns/op          37.58 MB/s
BenchmarkParsePointsTagsSorted2-8        2000000               847 ns/op          60.15 MB/s
BenchmarkParsePointsTagsSorted5-8        1000000              1033 ns/op          80.32 MB/s
BenchmarkParsePointsTagsSorted10-8       1000000              1445 ns/op          98.93 MB/s
BenchmarkParsePointsTagsUnSorted2-8      1000000              1047 ns/op          48.68 MB/s
BenchmarkParsePointsTagsUnSorted5-8      1000000              1044 ns/op          79.44 MB/s
BenchmarkParsePointsTagsUnSorted10-8      500000              2580 ns/op          55.42 MB/s
```

Benchamrks for this branch:

```
BenchmarkMarshal-8                        300000              5158 ns/op
BenchmarkParsePointNoTags-8              2000000               630 ns/op          36.45 MB/s
BenchmarkParsePointsTagsSorted2-8        2000000               849 ns/op          60.07 MB/s
BenchmarkParsePointsTagsSorted5-8        1000000              1041 ns/op          79.71 MB/s
BenchmarkParsePointsTagsSorted10-8       1000000              1397 ns/op         102.34 MB/s
BenchmarkParsePointsTagsUnSorted2-8      1000000              1036 ns/op          49.20 MB/s
BenchmarkParsePointsTagsUnSorted5-8      1000000              1044 ns/op          79.49 MB/s
BenchmarkParsePointsTagsUnSorted10-8      500000              2489 ns/op          57.44 MB/s
```

But all of the benchmarks just parse with precision of "n" so I have added new benchmarks that parse with precision 'n' and 'u' to show the difference. When parsing with a non 'n' precision there is an extra divide operation. 

```
BenchmarkParsePointWithPrecisionN-8      2000000               624 ns/op          36.81 MB/s
BenchmarkParsePointWithPrecisionU-8      2000000               625 ns/op          36.77 MB/s
```

Based on these benchmarks there is not a significant difference between precision of 'n' and precision of 'u'